In [1]:
import pandas as pd
import numpy as np
import re
from matplotlib import pyplot as plt

In [6]:
df=pd.read_csv("C:/Users/sylva/OneDrive/Bureau/barometre/avant_nettoyage.csv", sep=";", encoding='latin-1')

C:\Users\sylva\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (4,60,61,62,69) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [26]:
df

Commune Code_INSEE                Commune_cp  \
0                         Paris (75000)      75056                     Paris   
1      Conflans-Sainte-Honorine (78700)      78172  Conflans-Sainte-Honorine   
2                        Pessac (33600)      33318                    Pessac   
3                     Marseille (13000)      13055                 Marseille   
4                      Trumilly (60800)      60650                  Trumilly   
...                                 ...        ...                       ...   
68505                     Paris (75000)      75056                     Paris   
68506       Bruille-Saint-Amand (59199)      59114       Bruille-Saint-Amand   
68507               Montpellier (34000)      34172               Montpellier   
68508                     Paris (75000)      75056                     Paris   
68509                               NaN        NaN                       NaN   

       Dep     Cp           Nom_dep                  Nom_region Malvoyant  \
0       75  75000             Paris               Île-de-France       Non   
1       78  78700          Yvelines               Île-de-France       Non   
2       33  33600           Gironde          Nouvelle-Aquitaine       Non   
3       13  13000  Bouches-du-Rhône  Provence-Alpes-Côte d'Azur       Non   
4       60  60800              Oise             Hauts-de-France       Non   
...    ...    ...               ...                         ...       ...   
68505   75  75000             Paris               Île-de-France       Non   
68506   59  59199              Nord             Hauts-de-France       Non   
68507   34  34000           Hérault                   Occitanie       Non   
68508   75  75000             Paris               Île-de-France       Non   
68509  NaN    NaN               NaN                         NaN       Non   

      Deplacement_a_pied_vie_courante  Deplacement_a_pied_loisir  ...  \
0           Tous les jours ou presque  Tous les jours ou presque  ...   
1           Tous les jours ou presque  Tous les jours ou presque  ...   
2           Tous les jours ou presque  Tous les jours ou presque  ...   
3                      Presque jamais  Tous les jours ou presque  ...   
4                      Presque jamais  Tous les jours ou presque  ...   
...                               ...                        ...  ...   
68505       Tous les jours ou presque  Tous les jours ou presque  ...   
68506                  Presque jamais     1 à 3 fois par semaine  ...   
68507       Tous les jours ou presque     1 à 3 fois par semaine  ...   
68508       Tous les jours ou presque  Tous les jours ou presque  ...   
68509                             NaN                        NaN  ...   

      Dep_residence_OK  Cp_residence_OK  Nom_dep_residence_OK  \
0                   75            75000                 Paris   
1                   78            78700              Yvelines   
2                   33            33600               Gironde   
3                   13            13000      Bouches-du-Rhône   
4                   60            60800                  Oise   
...                ...              ...                   ...   
68505              NaN              NaN                   NaN   
68506              NaN              NaN                   NaN   
68507               34            34000               Hérault   
68508               75            75000                 Paris   
68509              NaN              NaN                   NaN   

          Nom_region_residence_OK  Indicateur_ressenti_global  \
0                   Île-de-France                         3,5   
1                   Île-de-France                           4   
2              Nouvelle-Aquitaine                 2,666666667   
3      Provence-Alpes-Côte d'Azur                 1,333333333   
4                 Hauts-de-France                 2,333333333   
...                           ...                         ...   
68505                         NaN

In [8]:
df.count()

Commune                             55664
Code_INSEE                          55664
Commune_cp                          55664
Dep                                 55664
Cp                                  55664
                                    ...  
Indicateur_securite                 50263
Indicateur_confort                  47291
Indicateur_effort_ville             45881
Indicateur_amenagements_services    45434
Indicateur_global                   51690
Length: 78, dtype: int64

In [ ]:
df

In [41]:
df_pascommune=df[df['Commune'].isna()]
for email in df_pascommune['Email_contact'].unique():
    taille=df_pascommune[df_pascommune['Email_contact']==email].shape
    if taille[0]>1 : print(taille)

(2, 78)
(2, 78)
(2, 78)
(3, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(4, 78)
(3, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(3, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(4, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(3, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(3, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)
(2, 78)


In [18]:
a=df[df['Commune'].isna()].count()

In [25]:
a[0:50]

Commune                                     0
Code_INSEE                                  0
Commune_cp                                  0
Dep                                         0
Cp                                          0
Nom_dep                                     0
Nom_region                                  0
Malvoyant                               11843
Deplacement_a_pied_vie_courante            15
Deplacement_a_pied_loisir                  15
Lieu_marche                                13
Deplacement_a_pied                         13
Agencement_voirie                          13
conflits_pietons                           13
Conducteurs_respectent                     12
Circulation_vehicules_motorises            12
Deplacements_pied_commune                  11
Marche_securite                            11
Securite_rues_services                     11
Securite_rues_residentielles               11
Securite_rejoindre_communes_voisines       11
Acces_pietons                     

In [24]:
a[50:]

Lors_de_vos_deplacements                 5
Principal_moyen_transport                0
Raisons_pas_deplacement_pied             0
Permis_de_conduire                       5
Possedez-vous                            5
Abonnement_transport_commun              5
Membre_association_marche                5
Membre_association_defense_pietons       5
Sexe_repondant                           5
Age_repondant                            5
Reception_resultats                      0
Reception_info_promoteurs_baro           0
Recuperation_email                    6212
Email_contact                         2286
Email_contact_malvoyant                 30
CommuneResidence_OK                      0
Code_INSEE_residence_OK                  0
Commune_cp_residence_OK                  0
Dep_residence_OK                         0
Cp_residence_OK                          0
Nom_dep_residence_OK                     0
Nom_region_residence_OK                  0
Indicateur_ressenti_global              13
Indicateur_

12846 personnes n'ont pas renseignés la commune, dont 11843 qui ont renseignés malvoyant, 6112 la récupération es emails, et moins de 15 par question

In [27]:
df_commune=df[df['Commune'].notna()]

In [30]:
df_commune.count()[:50]

Commune                                 55664
Code_INSEE                              55664
Commune_cp                              55664
Dep                                     55664
Cp                                      55664
Nom_dep                                 55656
Nom_region                              55656
Malvoyant                               55662
Deplacement_a_pied_vie_courante         53541
Deplacement_a_pied_loisir               53527
Lieu_marche                             53523
Deplacement_a_pied                      51676
Agencement_voirie                       51671
conflits_pietons                        51669
Conducteurs_respectent                  50994
Circulation_vehicules_motorises         50991
Deplacements_pied_commune               50986
Marche_securite                         50252
Securite_rues_services                  50249
Securite_rues_residentielles            50249
Securite_rejoindre_communes_voisines    50248
Acces_pietons                     

In [31]:
df_commune.count()[50:]

Lors_de_vos_deplacements              43372
Principal_moyen_transport               739
Raisons_pas_deplacement_pied            732
Permis_de_conduire                    43973
Possedez-vous                         43973
Abonnement_transport_commun           43972
Membre_association_marche             43231
Membre_association_defense_pietons    43231
Sexe_repondant                        43884
Age_repondant                         43885
Reception_resultats                   10962
Reception_info_promoteurs_baro        10962
Recuperation_email                    44702
Email_contact                         34454
Email_contact_malvoyant                 501
CommuneResidence_OK                   44522
Code_INSEE_residence_OK               44522
Commune_cp_residence_OK               44522
Dep_residence_OK                      44522
Cp_residence_OK                       44522
Nom_dep_residence_OK                  44517
Nom_region_residence_OK               44517
Indicateur_ressenti_global      

In [33]:
df_commune[df_commune['Deplacement_a_pied'].isna()].count()[0:50]

Commune                                 3988
Code_INSEE                              3988
Commune_cp                              3988
Dep                                     3988
Cp                                      3988
Nom_dep                                 3986
Nom_region                              3986
Malvoyant                               3988
Deplacement_a_pied_vie_courante         1865
Deplacement_a_pied_loisir               1851
Lieu_marche                             1847
Deplacement_a_pied                         0
Agencement_voirie                          0
conflits_pietons                           1
Conducteurs_respectent                     0
Circulation_vehicules_motorises            0
Deplacements_pied_commune                  0
Marche_securite                            0
Securite_rues_services                     0
Securite_rues_residentielles               0
Securite_rejoindre_communes_voisines       0
Acces_pietons                              0
Marche_sep

In [34]:
df_commune[df_commune['Deplacement_a_pied'].isna()].count()[50:]

Lors_de_vos_deplacements                 0
Principal_moyen_transport              727
Raisons_pas_deplacement_pied           726
Permis_de_conduire                     722
Possedez-vous                          722
Abonnement_transport_commun            722
Membre_association_marche                0
Membre_association_defense_pietons       0
Sexe_repondant                         719
Age_repondant                          719
Reception_resultats                    134
Reception_info_promoteurs_baro         134
Recuperation_email                    3854
Email_contact                         1862
Email_contact_malvoyant                 21
CommuneResidence_OK                    759
Code_INSEE_residence_OK                759
Commune_cp_residence_OK                759
Dep_residence_OK                       759
Cp_residence_OK                        759
Nom_dep_residence_OK                   759
Nom_region_residence_OK                759
Indicateur_ressenti_global               1
Indicateur_

3988 personnes qui n'ont répondu à aucune question de ressenti, dont 727 qui l'ont fait car ils ne marchent jamais.

In [45]:
df_commune[df_commune['Securite_rues_residentielles'].isna()].count()[:50]

Commune                                 5415
Code_INSEE                              5415
Commune_cp                              5415
Dep                                     5415
Cp                                      5415
Nom_dep                                 5413
Nom_region                              5413
Malvoyant                               5415
Deplacement_a_pied_vie_courante         3292
Deplacement_a_pied_loisir               3278
Lieu_marche                             3274
Deplacement_a_pied                      1427
Agencement_voirie                       1422
conflits_pietons                        1420
Conducteurs_respectent                   745
Circulation_vehicules_motorises          742
Deplacements_pied_commune                737
Marche_securite                            3
Securite_rues_services                     1
Securite_rues_residentielles               0
Securite_rejoindre_communes_voisines       0
Acces_pietons                              0
Marche_sep